### PR TITLE
fix: refactor allSettled to remove dependencies

### PR DESCRIPTION
**Pull request by KonstantinosKoulaxizis**
Remove allSettled package method and use a local one with the same functionality.

allSettled method was throwing `Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]`

### CODE REVIEW

**Comment by oscb** [comment]
@KonstantinosKoulaxizis Thanks for the contribution! I don't quite get a repro on this error but it seems like a fairly small change so if this fixes it let's do it. I refactored the implementation a bit and added tests.

**Review by silesky — APPROVED** [review]
Coincidence... I recently wrote the same polyfill =)
```ts
type AllSettledResult = {
  status: 'fulfilled' | 'rejected'
  reason?: Error
  value?: T
}

// promise.allSettled "polyfill"
export const allSettled = (
  promises: Promise[]
): Promise[]> => {
  return Promise.all(
    promises.map((promise) => {
      // Per spec, Promise.allSettled([Promise.reject('oops'), 2] ->
      // [{status: "rejected", reason: "oops"}, { "status": "fulfilled", "value": 2 }}
      return Promise.resolve(promise)
        .then(
          (value) =>
            ({
              value,
              status: 'fulfilled',
            } as AllSettledResult)
        )
        .catch(
          (reason) =>
            ({
              reason,
              status: 'rejected',
            } as AllSettledResult)
        )
    })
  )
}


```

File: packages/core/src/util.ts [hunk 1]
```diff
@@ -94,3 +94,36 @@ export const getPluginsWithReset = (timeline: Timeline) => {
 
   return eventPlugins;
 };
+
+type PromiseResult<T> =
```
Comment (silesky) [inline packages/core/src/util.ts:98]: This is an excellent typescript definition

**Pull request merged**